test/unit/unit_socket.c

```diff
@@ -39,13 +39,13 @@ START_TEST (test_socket)
     s = pico_socket_open(PICO_PROTO_IPV4, 99, NULL);
     fail_if(s != NULL, "Error got socket wrong parameters");
 
-    s = pico_socket_open(PICO_PROTO_IPV4, -109, NULL);
+    s = pico_socket_open(PICO_PROTO_IPV4, (uint16_t)-109, NULL);
```

Comment:
I removed the tests where a negative number is supplied for a uint*_t parameter (7ee1b3566e32ce4b292ab7b9cadd14833de6ed77). Is this okay for you guys?